app/src/main/java/com/duckduckgo/app/global/view/DaxDialog.kt

```diff
@@ -0,0 +1,147 @@
+    private val typingDelayInMs: Long = 20
+) : DialogFragment() {
+
+    private var onAnimationFinished: () -> Unit = {}
+    private var primaryCTAClickListener: () -> Unit = { dismiss() }
+    private var hideClickListener: () -> Unit = { dismiss() }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? =
+        inflater.inflate(R.layout.content_dax_dialog, container, false)
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val dialog = super.onCreateDialog(savedInstanceState)
+        val window = dialog.window
+        val attributes = window?.attributes
+
+        attributes?.gravity = Gravity.BOTTOM
+        window?.attributes = attributes
+        window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
+
+        return dialog
+    }
+
+    override fun onStart() {
+        super.onStart()
+        setDialog()
+        setListeners()
+        dialogText.startTypingAnimation(daxText, true, onAnimationFinished)
+    }
+
+    override fun onCancel(dialog: DialogInterface) {
+        dialogText.cancelAnimation()
+        super.onCancel(dialog)
+    }
+
+    fun onAnimationFinishedListener(onAnimationFinished: () -> Unit) {
+        this.onAnimationFinished = onAnimationFinished
+    }
+
+    fun setPrimaryCTAClickListener(clickListener: () -> Unit) {
+        primaryCTAClickListener = clickListener
+    }
+
+    fun setHideClickListener(clickListener: () -> Unit) {
+        hideClickListener = clickListener
+    }
+
+    fun startHighlightViewAnimation(targetView: View, duration: Long = 400, size: Float = 6f) {
+        val highlightImageView = addHighlightView(targetView, size)
+        val fadeInAnimation = ScaleAnimation(0f, 1f, 0f, 1f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f)
+        fadeInAnimation.duration = duration
+        fadeInAnimation.fillAfter = true
+        fadeInAnimation.interpolator = OvershootInterpolator(3f)
+        highlightImageView.startAnimation(fadeInAnimation)
+    }
+
+    private fun setListeners() {
+        hideText.setOnClickListener {
+            dialogText.cancelAnimation()
+            hideClickListener()
+        }
+
+        primaryCta.setOnClickListener {
+            dialogText.cancelAnimation()
+            primaryCTAClickListener()
+        }
+
+        if (dismissible) {
+            dialogContainer.setOnClickListener {
+                dialogText.cancelAnimation()
+                dismiss()
+            }
+        }
+    }
+
+    private fun setDialog() {
+        dialog?.window?.setLayout(ConstraintLayout.LayoutParams.MATCH_PARENT, ConstraintLayout.LayoutParams.MATCH_PARENT)
+        hiddenText.text = daxText
+        primaryCta.text = buttonText
+        dialogText.typingDelayInMs = typingDelayInMs
+    }
+
+    private fun addHighlightView(targetView: View, size: Float): View {
+        val highlightImageView = ImageView(requireContext())
```

Comment:
We have `requireContext()` here, and `requireActivity()` further down. I feel like we could wrap all of this into a null-safe version. We shouldn't find that `activity` is null here (i don't think) but if it is, it would be nice to not crash the app because of it.

Reply:
Done!